tests/cupy_tests/core_tests/test_ndarray.py

```diff
@@ -39,44 +40,77 @@ def wrap_take(array, *args, **kwargs):
     return array.take(*args, **kwargs)
 
 
-@testing.parameterize(
-    {'arg': None, 'shape': ()},
-    {'arg': 3, 'shape': (3,)},
-)
 @testing.gpu
 class TestNdarrayInit(unittest.TestCase):
 
-    def test_shape(self):
-        a = core.ndarray(self.arg)
-        self.assertTupleEqual(a.shape, self.shape)
+    def test_shape_none(self):
+        a = cupy.ndarray(None)
+        assert a.shape == ()
 
+    def test_shape_int(self):
+        a = cupy.ndarray(3)
+        assert a.shape == (3, )
 
-@testing.gpu
-class TestNdarrayOrder(unittest.TestCase):
+    def test_memptr(self):
+        a = cupy.arange(6).astype(numpy.float32).reshape((2, 3))
+        memptr = a.data
+
+        b = cupy.ndarray((2, 3), numpy.float32, memptr)
+        testing.assert_array_equal(a, b)
+
+        b += 1
+        testing.assert_array_equal(a, b)
+
+    def test_memptr_with_strides(self):
+        a = cupy.arange(12).astype(numpy.float32).reshape((2, 6))[::-1, ::2]
+        memptr = a.data
+
+        b = cupy.ndarray((2, 3), numpy.float32, memptr, strides=a.strides)
+        testing.assert_array_equal(a, b)
+        assert a.strides == b.strides
+
+        b += 1
+        testing.assert_array_equal(a, b)
+
+    def test_strides(self):
+        a = cupy.ndarray((2, 3), numpy.float32, strides=(8, 4))
+        assert a.strides == (8, 4)
+
+        a[:] = 1
+        a[0, 2] = 4
+        assert float(a[1, 0]) == 4
+
+    def test_strides_is_given_and_order_is_ignored(self):
+        a = cupy.ndarray((2, 3), numpy.float32, strides=(8, 4), order='C')
+        assert a.strides == (8, 4)
 
-    shape = (2, 3, 4)
+    def test_strides_is_given_but_order_is_invalid(self):
+        with pytest.raises(TypeError):
+            cupy.ndarray((2, 3), numpy.float32, strides=(8, 4), order='!')
```

Comment:
ditto

Reply:
I think this case is OK because the test just checks raised error.